tools/java/org.hl7.fhir.tools.core/src/org/hl7/fhir/tools/implementations/ember/EmberModel.java

```diff
@@ -0,0 +1,359 @@
+        fileBlock.ln();
+        fileBlock.ln();
+        fileBlock.ln("import DS from 'ember-data';");
+        fileBlock.ln();
+        fileBlock.ln();
+
+    }
+
+    private void generateResourceStruct(GenBlock fileBlock) {
+        fileBlock.bs("var " + name + " = DS.Model.extend({");
+        boolean firstLine = true;
+        for (ElementDefn elementDefinition : getRootDefinition().getElements()) {
+            generateFields(fileBlock, elementDefinition, firstLine);
+            firstLine = false;
+        }
+        fileBlock.es("});");
+        fileBlock.ln("export default " + name + ";");
+    }
+
+    private void generateComponentStructs(GenBlock fileBlock) throws Exception {
+        for (ElementDefn elementDefinition : getRootDefinition().getElements()) {
+            generateComponentStruct(fileBlock, elementDefinition);
+        }
+    }
+
+    // private void generateCustomMarshallersAndUnMarshallers(GenBlock fileBlock) throws Exception {
+    //     if (definitions.getResources().containsKey(name)) {
+    //         ST st = templateGroup.getInstanceOf("generic_resource_marshaller.go");
+    //         st.add("Name", name);
+    //         fileBlock.ln(st.render());
+    //     }
+    //
+    //     generateCustomUnmarshaller(name, getRootDefinition(), fileBlock);
+    // }
+    //
+    // private void generateCustomUnmarshaller(String structName, ElementDefn elementDefn, GenBlock fileBlock) {
+    //     ArrayList<String> resourceFields = new ArrayList<String>();
+    //     for (ElementDefn elementDefinition : elementDefn.getElements()) {
+    //         for (TypeRef typeRef : elementDefinition.getTypes()) {
+    //             if ("Resource".equals(typeRef.getName()) || "Resource".equals(typeRef.getResolvedTypeName())) {
+    //                 resourceFields.add(getFieldName(elementDefinition, typeRef));
+    //             }
+    //         }
+    //     }
+    //
+    //     if (!resourceFields.isEmpty()) {
+    //         ST st = templateGroup.getInstanceOf("generic_resource_unmarshaller.go");
+    //         st.add("Name", structName);
+    //         st.add("LowerName", lowercase(structName));
+    //         st.add("Fields", resourceFields);
+    //         fileBlock.ln(st.render());
+    //     }
+    // }
+
+    private void generateFields(GenBlock block, ElementDefn elementDefinition) {
+      generateFields(block,elementDefinition, true);
+    }
+
+    private void generateFields(GenBlock block, ElementDefn elementDefinition, boolean isFirst) {
+        if (isComponent(elementDefinition)) {
+            String fieldName = getFieldName(elementDefinition, null);
+            String structName = dasherize(getComponentStructName(elementDefinition));
+            Integer card = elementDefinition.getMaxCardinality();
+            String modifier = card == null || card > 1 ? "DS.hasMany" : "DS.belongsTo";
+            if( ! isFirst ){
+              block.nl(",");
+            }
+            isFirst = false;
```

Comment:
Re-assigning `isFirst` to false here doesn't do anything.  This isn't in a for loop (like the one below), so this particular re-assignment of `isFirst` will never actually be used.  Keep it in the for loop below though.